core/src/main/java/com/linecorp/armeria/server/throttling/ThrottlingAcceptHandler.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * A callback which is invoked to handle accepted (successful) requests indicated by {@link ThrottlingStrategy}.
+ *
+ * @param <I> the type of incoming {@link Request}. Must be {@link HttpRequest} or {@link RpcRequest}.
+ * @param <O> the type of outgoing {@link Response}. Must be {@link HttpResponse} or {@link RpcResponse}.
+ *
+ * @see ThrottlingServiceBuilder#onAcceptedRequest(ThrottlingAcceptHandler)
+ */
+@FunctionalInterface
+public interface ThrottlingAcceptHandler<I extends Request, O extends Response> {
```

Comment:
Just wondering, any thoughts on

- Removing accept handler (what can we expect other than just running the delegate?)
- Remove `delegate` from reject handler (what can we expect other than not running delegate?)

I think it's fine to have them too but it does add some complexity which might not ever be needed.

Reply:
> Remove delegate from reject handler (what can we expect other than not running delegate?)
Perhaps delegating and then just counting?

Reply:
I'd expect anything counting-related would be in the strategy. But not sure.

Reply:
Kept the way it is, similar to AuthService.

Reply:
Looking at `AuthService` was mainly for inspiration, and may be good to follow as is, or maybe can be tweaked to improve the API here it's all case-by-case. I think it may make sense to remove delegate from `AuthService` too after a second look, though that's another topic.

Here, there is even more to worry about since the strategy will actually be counting and making throttling decisions. So an accept handler that doesn't call the delegate stands to mess up the strategy quite easily, as would a failure handler that does call it. Decoupling the execution of a service with the generation of an error response seems cleaner here unless there are other concerns.

Anyways, the goal here is not to follow `AuthService` pattern word for word but to come up with a clean API for throttling.

Reply:
I’m slightly in favor of keeping the API as it is. A handle might want to insert some response header, schedule some task or increase a counter on accept/reject.

Reply:
Pretty sure the strategy can do all of these too - but it's ok to keep it anyways. Let's make sure the javadoc is clear about the expected use of delegate, especially on failure where it will usually cause a problem.